my-webxr-app/src/components/DataPointMenu.tsx

```diff
@@ -14,12 +14,20 @@ export default function DataPointMenu(
 
   return (
     <Billboard visible={selectedDataPoint != null} {...billboardProps}>
-      <Plane args={[1.75, 0.25]}>
-        <Text fontSize={0.1} color={"black"}>
-          Here are data point #
-          {selectedDataPoint == null ? "-" : selectedDataPoint} properties!
-        </Text>
-      </Plane>
+        <Plane args={[1.25, 0.75]}>
+
+                <Text fontSize={0.075} color={"black"}>
+                    Here are data point # {selectedDataPoint == null ? "-" : selectedDataPoint.id} properties!
```

Comment:
Since we already have the condition `visible={selectedDataPoint != null}` in the BillBoard element (the parent parent element of Text), we are sure that `selectedDataPoint == null` can't never happen here, right?

Reply:
I think we have to check if selectedDataPoint is null inside the Text element if all fields are not initialized when the data point object is not null. We can also create a variable outside to check the condition for less code redundancy:

Line 14: 
const dataPoint = selectedDataPoint || null;
Before:
Here are data point # {selectedDataPoint == null ? "-" : selectedDataPoint.id} properties!
After:
Here are data point # {dataPoint?.id ?? "-"} properties!

What do you think?

Reply:
Yeah, that's a bit of a thinker. I first thought that it was unnecessary too, but I believe we need a form of it since that code still runs whether the billboard is visible or not—it's not like an `if` statement; just visible or not visible and the Billboard element still gets created.

I'd never seen `??` used before, but I think it would work here to make things a bit more concise. Something like this:
```ts
Here are data point # {selectedDataPoint?.id ?? "-"} properties!
```
Would you need the extra variable?

Reply:
No you actually wouldn't need the extra variable, good catch! I've made that block of code more concise now, and null is only being checked in the parent element.